### ISSUE TITLE
Child TextBundle render in bottom left corner if parent is Display::None

### ISSUE THREAD

**Issue by aspin** [bevyengine/bevy#1135]
**Bevy version**

Checked on both 0.4 and 61ce3f7.

**Operating system & version**

macOS Catalina

**What you did**

Added a child `TextBundle` to a root `NodeBundle`, then set the root node to not be displayed.

Here's a full script:

```rust
use bevy::prelude::*;

fn main() {
    App::build()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup.system())
        .run();
}

fn setup(
    commands: &mut Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<ColorMaterial>>,
) {
    commands
        // ui camera
        .spawn(CameraUiBundle::default())
        // root node
        .spawn(NodeBundle {
            style: Style {
                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
                justify_content: JustifyContent::SpaceBetween,
                display: Display::None,
                ..Default::default()
            },
            material: materials.add(Color::NONE.into()),
            ..Default::default()
        })
        .with_children(|parent| {
            parent.spawn(TextBundle {
                style: Style {
                    position_type: PositionType::Absolute,
                    position: Rect {
                        bottom: Val::Px(0.),
                        right: Val::Px(0.),
                        ..Default::default()
                    },
                    ..Default::default()
                },
                text: Text {
                    value: "SAMPLE TEXT".to_string(),
                    font: asset_server.load("fonts/FiraSans-Bold.ttf"),
                    style: TextStyle {
                        color: Color::WHITE,
                        font_size: 30.0,
                        ..Default::default()
                    },
                    ..Default::default()
                },
                ..Default::default()
            });
        });
}
```

**What you expected to happen**

`TextBundle` should not have been shown on the screen at all.

**What actually happened**

`TextBundle` rendered to the bottom left corner of the screen.

**Additional information**
Unexpected:
![unexpected result](https://user-images.githubusercontent.com/5728920/102964065-52433080-44b0-11eb-8ab8-aceb2668cdc6.png)

If root is displayed:
![if root is hown](https://user-images.githubusercontent.com/5728920/102964047-48213200-44b0-11eb-8a4d-faff441c7f22.png)

**Comment by kumorig** [comment]
https://github.com/bevyengine/bevy/issues/838 (May or may not be related)

**Comment by aspin** [comment]
I believe that this is a regression from 0.3. Display::None worked to hide child elements in 0.3 (I think you pointed that out to me in discord), while is_visible didnt.

**Comment by ristew** [comment]
I'm still having this issue, interesting is the computed node size is still zero so I can use this as a workaround: 
```
fn issue_1135_system(
    mut text_style_visible_query: Query<(&Text, &Node, &mut Visible)>,
) {
    for (text, node, mut visible) in text_style_visible_query.iter_mut() {
        if node.size == Vec2::ZERO {
            visible.is_visible = false;
        } else {
            visible.is_visible = true;
        }
    }
}

```

**Comment by swwind** [comment]
Problem still exists in 0.5.0.

**Comment by rparrett** [comment]
This was fixed by #3312.
